packages/react/src/wallet/infrastructure/SignerFactory.ts

```diff
@@ -37,6 +39,16 @@ export class SignerFactory implements ISignerFactory {
     }
 
     if (chainType) {
+      try {
+        await signer.getChainId();
```

Comment:
when it's possible for that part of code to fail? 
Wondering if we can simplify the code to not call the `signer.getChainId` twice and to not return `WalletConnectionError(WalletConnectionErrorReason.INCORRECT_CHAIN)` twice (here and inside `switchChain`).

Reply:
ah yep you're right that would be simpler